shared/src/types/ibc.rs

```diff
@@ -600,3 +601,549 @@ impl ConnectionOpenConfirmData {
         .map_err(Error::DecodingError)
     }
 }
+
+/// Data to initialize a channel
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
+pub struct ChannelOpenInitData {
+    /// The port ID
+    port_id: String,
+    /// The order feature of the channel
+    order: String,
+    /// The counterparty
+    counterparty: Vec<u8>,
+    /// The connection hops
+    connection_hops: Vec<String>,
+    /// The version
+    version: String,
+}
+
+impl ChannelOpenInitData {
+    /// Returns the data to initalize a channel
+    pub fn new(
+        port_id: PortId,
+        order: Order,
+        counterparty: ChanCounterparty,
+        connection_hops: Vec<ConnectionId>,
+        version: String,
+    ) -> Self {
+        let port_id = port_id.as_str().to_owned();
+        let order = order.as_str().to_owned();
```

Comment:
These can use the `Display` instances directly
```suggestion
        let port_id = port_id.to_string();
        let order = order.to_string();
```